### PR TITLE
League of Memories: New changes report

### PR DESCRIPTION
**Pull request by Deudly**
* Added the necessary functions for passives to update
* Changing the way Recall is handeled

Note that [new `Recall.cs` file can be found in our Spells repository ](https://github.com/LeagueofMemories/LeagueOfMemories-Default/blob/master/Champions/Global/Recall.cs). Unfortunatly, this PR cannot update your submodule since we are not using it

### CODE REVIEW

File: GameServerLib/Logic/GameObjects/AttackableUnits/AI/Champion.cs [hunk 1]
```diff
@@ -67,6 +67,12 @@ public class Champion : ObjAIBase
             }
             Spells[4] = new Spell(this, clientInfo.SummonerSkills[0], 4);
             Spells[5] = new Spell(this, clientInfo.SummonerSkills[1], 5);
+
+            for(short i = 0; i < 6; i++)
+            {
+                Spells[(byte)(i + 6)] = new Spell(this, "", (byte)(i + 6));
```
Comment (FurkanS1821) [inline GameServerLib/Logic/GameObjects/AttackableUnits/AI/Champion.cs]: There's exactly what you would want to use as an empty spell: `BaseSpell`.
Riot uses these for non-champion objects, and I believe going that way is better than using an empty string.

File: GameServerLib/Logic/GameObjects/AttackableUnits/AI/Champion.cs [hunk 2]
```diff
@@ -211,7 +225,15 @@ public Vector2 GetRespawnPosition()
...
         {
-            return Spells[slot];
+            if (Spells.ContainsKey(slot))
+            {
+                return Spells[slot];
+            }
+            else
+            {
+                ApiFunctionManager.LogInfo("There is no spell in slot " + slot + ". Using empty spell in its place.");
+                return new Spell(this, "", slot);
```
Comment (FurkanS1821) [inline GameServerLib/Logic/GameObjects/AttackableUnits/AI/Champion.cs]: There's exactly what you would want to use as an empty spell: BaseSpell.
Riot uses these for non-champion objects, and I believe going that way is better than using an empty string.

File: GameServerLib/Logic/GameObjects/AttackableUnits/AI/Champion.cs [hunk 3]
```diff
@@ -245,14 +267,15 @@ public Spell LevelUpSpell(short slot)
         public override void update(float diff)
         {
             base.update(diff);
-            if (this.isMovementUpdated()) {
-                if (isRecalling)
+			
+			if (this.isMovementUpdated()) {
```
Comment (FurkanS1821) [inline GameServerLib/Logic/GameObjects/AttackableUnits/AI/Champion.cs]: indentation

File: GameServerLib/Logic/GameObjects/AttackableUnits/AI/Champion.cs [hunk 4]
```diff
@@ -329,18 +354,20 @@ public void Respawn()
...
         }
 
-	    public void Recall(ObjAIBase owner,float timer)
+	    public void Recall(ObjAIBase owner,float timer=8.0f)
         {
-            var spawnPos = GetRespawnPosition();
-            isRecalling = true;
-            visualBuff = ApiFunctionManager.AddBuffHUDVisual("Recall", 8.0f, 1, this);
-            addParticle = ApiFunctionManager.AddParticleTarget(this, "TeleportHome.troy", this);
+			var spawnPos = GetRespawnPosition();
```
Comment (FurkanS1821) [inline GameServerLib/Logic/GameObjects/AttackableUnits/AI/Champion.cs]: indentation

File: GameServerLib/Logic/GameObjects/AttackableUnits/AI/Champion.cs (hunk 3 above)
Comment (FurkanS1821) [inline GameServerLib/Logic/GameObjects/AttackableUnits/AI/Champion.cs]: \+ bracket should be one line below

File: GameServerLib/Logic/GameObjects/Spells/Spell.cs [hunk 5]
```diff
@@ -69,7 +69,10 @@ public Spell(Champion owner, string spellName, byte slot)
             _scriptEngine = Program.ResolveDependency<CSharpScriptEngine>();
 
             //Set the game script for the spell
-            spellGameScript = _scriptEngine.CreateObject<GameScript>("Spells", spellName);
+            if (spellName != "")
```
Comment (FurkanS1821) [inline GameServerLib/Logic/GameObjects/Spells/Spell.cs]: `!spellName.Equals("")`, or even better: `!string.IsNullOrEmpty(spellName)`

here is why: https://stackoverflow.com/a/814881

**Review by FurkanS1821 — CHANGES_REQUESTED** [review]

**Comment by HarmfulBreeze** [comment]
Also see my comments on #563

**Review by HarmfulBreeze — CHANGES_REQUESTED** [review]
Fix conflicts and the issues Furkan mentioned.
Also please give actual names to your commits, not just "commit" or "update"
Besides that, not much else to say

File: .travis.yml [hunk 6]
```diff
@@ -2,13 +2,9 @@ sudo: true
...
 dist: trusty
-mono: none
+mono: latest
+
 install:
-    - sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF
-    - echo "deb http://download.mono-project.com/repo/ubuntu trusty main" | sudo tee /etc/apt/sources.list.d/mono-official.list
-    - sudo apt-get update
-    - sudo apt-get -y install mono-devel
-    - sudo apt-get install nuget
```
Comment (HarmfulBreeze) [inline .travis.yml]: Are you sure it still works without installing the nuget package?

**Review by HarmfulBreeze — COMMENTED** [review]
Also too many changes! God!

**Comment by Deudly** [comment]
Closing this PR since it's useless

**Pull request closed without merging**